core/src/components/select/test/basic/select.e2e.ts

```diff
@@ -153,7 +155,39 @@ test.describe('select: ionChange', () => {
     await radioButtons.nth(0).click();
 
     await ionChange.next();
-    expect(ionChange).toHaveReceivedEventDetail({ value: 'apple', event: { isTrusted: true } });
+    expect(ionChange).toHaveReceivedEventDetail({ value: 'apple' });
+    expect(ionChange).toHaveReceivedEventTimes(1);
+  });
+
+  test('should fire ionChange when confirming multiple values from a popover', async ({ page }) => {
```

Comment:
I can fill this information back into the main PR description, but this problem effects single selection for `ion-select`, regressing here: https://github.com/ionic-team/ionic-framework/pull/23474

The root problem was that we had both a click handler on the radio and a `@Listen` decorator for `ionChange`, both invoking the callback handler for the popover options. When a radio is selected, it would fire the click handler directly attached to the element and then the `ion-radio-group` would fire an `ionChange` event, additionally executing the callback handler (again).

This proposed fix works, because it maintains the click handler on the radio and moves the listener behavior directly onto the click handler for the checkbox (multiple selection). The `setChecked` implementation is only used/necessary for multiple selection with the checkboxes.

The tests you are commenting on are within a `test.describe` block, that skips execution for RTL and iOS. 

```ts
test.describe('select: ionChange', () => {
  test.beforeEach(({ skip }) => {
    skip.rtl();
    skip.mode('ios', 'ionChange has a consistent behavior across modes');
  });

  ...
});
```

One thing that I did not think to test originally that I will run through, is making sure keyboard interactions behave the same as well for selections. If not, there may be some rework to use the `ionChange` event instead of a click handler.